src/iceberg/transform_function.cc

```diff
@@ -51,6 +60,57 @@ Result<ArrowArray> BucketTransform::Transform(const ArrowArray& input) {
   return NotImplemented("BucketTransform::Transform");
 }
 
+Result<std::optional<Literal>> BucketTransform::Transform(const Literal& literal) {
+  assert(literal.type() == source_type());
+  if (literal.IsBelowMin() || literal.IsAboveMax()) {
+    return InvalidArgument(
+        "Cannot apply bucket transform to literal with value {} of type {}",
+        literal.ToString(), source_type()->ToString());
+  }
+  int32_t hash_value = 0;
+  switch (source_type()->type_id()) {
```

Comment:
Since we don't actually care about the type, you could `std::visit` the `value()` instead?

Reply:
Seems `literal.type() == source_type()` is enforced?

Reply:
Ok, I will try this later, thanks for the input.